drivers/modem/ublox-sara-r4.c

```diff
@@ -190,6 +212,86 @@ static int modem_atoi(const char *s, const int err_value,
 	return ret;
 }
 
+#if defined(CONFIG_MODEM_UBLOX_SARA_AUTODETECT_APN)
+
+const char *modem_sim_profiles =
+	CONFIG_MODEM_UBLOX_SARA_AUTODETECT_APN_PROFILES;
+
+int find_apn(char *apn, int apnlen, const char *profiles, const char *imsi)
+{
+	int rc = -1;
+
+	/* try to find a match */
+	char *s = strstr(profiles, imsi);
+
+	if (s) {
+		/* find the assignment operator preceding the match */
+		while (s >= profiles && !strchr("=", *s)) {
+			s--;
+		}
+		/* find the apn preceding the assignment operator */
+		while (s >= profiles && strchr(" =", *s)) {
+			s--;
+		}
+
+		char *eos = s+1;
```

Comment:
Introduce the variable at the beginning of the block, not middle.

Reply:
done